lib/matplotlib/_mathtext.py

```diff
@@ -2595,3 +2599,20 @@ def auto_delim(self, s, loc, toks):
             # if "mid" in toks ... can be removed when requiring pyparsing 3.
             toks["mid"].asList() if "mid" in toks else [],
             toks["right"])
+
+    def boldsymbol(self, s, loc, toks):
+        self.push_state()
+        state = self.get_state()
+        state.font = 'bfit'
+        hlist = []
+        name = toks["value"]
+        for c in name:
+            if isinstance(c, Char):
+                c.font = 'bfit'
+                c._update_metrics()
+                hlist.append(c)
+            else:
+                hlist.append(c)
+        self.pop_state()
```

Comment:
Didn't really check, but does that work with nesting?  (i.e.`\boldsymbol{\mathnormal{foo}}` -- the inner one should win)

Reply:
The `check_figures_equal` test in the PR currently gives this output:
![test_boldsymbol png](https://user-images.githubusercontent.com/35831931/235445702-8cf1234b-bf6c-43ed-b193-f61dbf69acd5.png)